src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    coords.x === undefined
+      ? this.coords = {
+        x: 0,
+        y: 0,
+      }
+      : this.coords = coords;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    coords.z === undefined
+      ? this.coords.z = 0
+      : this.coords.z = coords.z;
```

Comment:
U can write it shorter 
```suggestion
    this.coords.z = coords.z || 0;
```

Reply:
Agree here too